samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,175 @@
+            IDataView testDataView = mlContext.Data.LoadFromTextFile<RankingData>(TestDatasetPath, hasHeader: true, separatorChar: '\t');
+
+            // STEP 2: Display first few rows of training data
+            ConsoleHelper.ShowDataViewInConsole(mlContext, trainDataView);
+
+            // STEP 3: Initialize our user-defined progress handler that AutoML will 
+            // invoke after each model it produces and evaluates.
+            var progressHandler = new RankingExperimentProgressHandler();
+
+            // STEP 4: Run AutoML ranking experiment
+            ConsoleHelper.ConsoleWriteHeader("=============== Running AutoML experiment ===============");
+            Console.WriteLine($"Running AutoML ranking experiment for {ExperimentTime} seconds...");
+
+            ExperimentResult<RankingMetrics> experimentResult = mlContext.Auto()
+                .CreateRankingExperiment(ExperimentTime)
+                .Execute(trainDataView, progressHandler: progressHandler);
+
+            // Print top models found by AutoML
+            Console.WriteLine();
+            PrintTopModels(experimentResult);
+
+            // STEP 5: Evaluate the model and print metrics
+            ConsoleHelper.ConsoleWriteHeader("=============== Evaluating model's accuracy with test data ===============");
+            RunDetail<RankingMetrics> bestRun = experimentResult.BestRun;
+
+            ITransformer trainedModel = bestRun.Model;
+            var predictions = trainedModel.Transform(testDataView);
+
+            // Get a row for single prediction
+            sampleRankingData = mlContext.Data.CreateEnumerable<RankingData>(mlContext.Data.TakeRows(testDataView, 1), reuseRowObject: false)
+                .FirstOrDefault();
+
+            var metrics = mlContext.Ranking.Evaluate(predictions);
+
+            ConsoleHelper.PrintRankingMetrics(bestRun.TrainerName, metrics);
+
+            // STEP 6: Save/persist the trained model to a .ZIP file
+            mlContext.Model.Save(trainedModel, trainDataView.Schema, ModelPath);
+
+            Console.WriteLine("The model is saved to {0}", ModelPath);
+
+            return trainedModel;
+        }
+
+        private static void TestSinglePrediction(MLContext mlContext)
+        {
+            ConsoleHelper.ConsoleWriteHeader("=============== Testing prediction engine ===============");
+
+
+            ITransformer trainedModel = mlContext.Model.Load(ModelPath, out var modelInputSchema);
+            Console.WriteLine($"=============== Loaded Model OK  ===============");
+
+            // Create prediction engine related to the loaded trained model
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<RankingData, RankingPrediction>(trainedModel, inputSchema: modelInputSchema);
+
+            var predictionResult = predictionEngine.Predict(sampleRankingData);
```

Comment:
Showing off ranked results of a single query would be more beneficial. 

I think this is done as:
* Get the testset's first row's `GroupID`
* Filter the IDataView to rows with that `GroupID` <br/>(or just know the number of rows in the first group)
* Score those rows
* Sort the rows by `score` (descending)
* Display rows, now in ranked order

Reply:
I think that's what they do [here](https://github.com/dotnet/machinelearning-samples/blob/master/samples/csharp/getting-started/Ranking_Web/WebRanking/Program.cs#L193) as well, so I used that as an example. Let me know if it's not correct, though.

Reply:
Quite so. The other ranking sample already had exactly what I was thinking of. Thanks for locating it and reusing here.

#RESOVLED